src/python/pants/backend/python/tasks/python_binary_create.py

```diff
@@ -126,8 +130,27 @@ def _create_binary(self, binary_tgt, results_dir):
       # Dump everything into the builder's chroot.
       for tgt in source_tgts:
         dump_sources(builder, tgt, self.context.log)
+
+      # Handle locally-built python distribution dependencies.
+      local_dist_req_libs = []
+      built_dists = self.context.products.get_data(BuildLocalPythonDistributions.PYTHON_DISTS)
+      if built_dists:
+        local_dist_req_libs = build_req_libs_provided_by_setup_file(self.context,
+                                                                    built_dists,
+                                                                    self.__class__.__name__)
+      req_tgts = local_dist_req_libs + req_tgts
```

Comment:
how about killing the `local_dist_req_libs` temp var here in favor of direct insertion/extension/replacement against `req_tgts`?

```
built_dists = ...
if built_dists:
  req_tgts = build_req_libs_provided_by_setup_file(...) + req_tgts
```

Reply:
✅